ibm-watson-workitem/src/main/java/org/jbpm/process/workitem/ibm/watson/ClassifyImageWorkitemHandler.java

```diff
@@ -0,0 +1,132 @@
+import com.ibm.watson.developer_cloud.visual_recognition.v3.model.ErrorInfo;
+import org.jbpm.document.Document;
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "ClassifyImageDefinitions.wid", name = "IBMWatsonClassifyImage",
+        displayName = "IBMWatsonClassifyImage",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.ibm.watson.ClassifyImageWorkitemHandler()",
+        parameters = {
+                @WidParameter(name = "ImageToClassify")
+        },
+        results = {
+                @WidResult(name = "Classification")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "com.ibm.watson.developer_cloud", artifact = "java-sdk", version = "4.0.0")
+        })
+public class ClassifyImageWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClassifyImageWorkitemHandler.class);
+    private static final String RESULT_VALUE = "Classification";
+    private VisualRecognition service;
+
+    private String apiKey;
+
+    public ClassifyImageWorkitemHandler(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+
+        Document classificationImage = (Document) workItem.getParameter("ImageToClassify");
+
+        Map<String, Object> widResults = new HashMap<String, Object>();
+
+        if (classificationImage != null) {
+
+            try {
+                if(service == null) {
+                    VisualRecognition service = new VisualRecognition(VisualRecognition.VERSION_DATE_2016_05_20);
+                    service.setApiKey(apiKey);
+                }
+
+                ByteArrayInputStream imageStream = new ByteArrayInputStream(classificationImage.getContent());
+
+                ClassifyOptions classifyOptions = new ClassifyOptions.Builder()
+                        .imagesFile(imageStream)
+                        .imagesFilename(classificationImage.getName())
+                        .parameters("{\"owners\": [\"me\"]}")
+                        .build();
+                ClassifiedImage result = service.classify(classifyOptions).execute().getImages().get(0);
+                if (result.getError() != null) {
+                    ErrorInfo errorInfo = result.getError();
+                    logger.error("Error classifying image: " + errorInfo.getDescription());
+                    workItemManager.completeWorkItem(workItem.getId(),
```

Comment:
should we rather fail here? otherwise users won't notice it failed at calling the service...